client_cli/src/main.rs

```diff
@@ -458,6 +461,87 @@ mod domain {
                 .wrap_err("Failed to transfer domain")
         }
     }
+
+    mod metadata {
+        use iroha_client::data_model::{self, domain::DomainId};
+
+        use super::*;
+
+        /// A value wrapper that can be parsed from CLI arguments
+        #[derive(Debug, Clone)]
+        pub struct MetadataValue(data_model::Value);
+
+        impl FromStr for MetadataValue {
+            type Err = Error;
+
+            fn from_str(s: &str) -> Result<Self> {
+                let deser_err_msg = format!("Failed to deserialize `{s}` into value.");
+                let metadata: Value = json5::from_str(s).wrap_err(deser_err_msg)?;
+                Ok(Self(metadata))
+            }
+        }
+
+        /// Edit domain subcommands
+        #[derive(Debug, Clone, clap::Subcommand)]
+        pub enum Args {
```

Comment:
Getting metadata would also be convinient

Reply:
I think metadata is shown when you use domain READ operations.